lib/mocha.js

```diff
@@ -143,7 +144,9 @@ Mocha.prototype.reporter = function(reporter, reporterOptions){
   } else {
     reporter = reporter || 'spec';
     var _reporter;
-    try { _reporter = require('./reporters/' + reporter); } catch (err) {}
+    // Try to load a built-in reporter.
+    if (reporters[reporter]) _reporter = reporters[reporter];
+    // Try to load reporters from process.cwd() and node_modules
```

Comment:
My thought was that Browserify's static requires would prevent hot loads here in the case that someone would want to load a custom reporter in the browser, but now that I'm thinking about it that might not be a problem? Not sure, haven't used Browserify for this specific scenario.

Mostly grasping at straws trying to think of a breaking use case because I haven't ever loaded a custom reporter in the browser, but that doesn't necessarily mean other people haven't.